packages/sanity/src/core/bundles/util/util.ts

```diff
@@ -12,6 +12,34 @@ export function getBundleSlug(documentId: string): string {
   return version
 }
 
+/**
+ * @beta
+ * @param documentId - The document id, e.g. `my-document-id` or `drafts.my-document-id` or `summer.my-document-id`
+ * @param perspective - The current perspective, e.g. `bundle.summer` or undefined, it can be obtained from `useRouter().stickyParams.perspective`
+ * @returns boolean - `true` if the document is in the current perspective.
+ * e.g:
+ * - document: `summer.my-document-id`, perspective: `bundle.summer` : **true**
+ * - document: `my-document-id`, perspective: `bundle.summer` : **false**
+ * - document: `summer.my-document-id`perspective: `bundle.winter` : **false**
+ * - document: `summer.my-document-id`, perspective: `undefined` : **false**
+ * - document: `my-document-id`, perspective: `undefined` : **true**
+ * - document: `drafts.my-document-id`, perspective: `undefined` : **true**
+ */
+export function getDocumentIsInPerspective(
+  documentId: string,
+  perspective: string | undefined,
+): boolean {
+  const bundleSlug = getBundleSlug(documentId)
+
+  if (!perspective) return bundleSlug === 'Published' || bundleSlug === 'drafts'
```

Comment:
is there a reason for the inconsistency in casing here?

Reply:
Yes, [getBundleSlug](https://github.com/sanity-io/sanity/blob/corel-74/packages/sanity/src/core/bundles/util/util.ts#L10-L11 )  returns "Published" when the document id is just `foo` and it will return "drafts" if the document id is `drafts.foo`
```
export function getBundleSlug(documentId: string): string {
  if (documentId.indexOf('.') === -1) return 'Published'
  const version = documentId.slice(0, documentId.indexOf('.'))
  return version
}
```

Reply:
That doesn't seem safe to me. It's [technically possible](https://www.sanity.io/docs/ids#10abf7adf05a) to create a document with the id `Published.foo`. I suggest we refactor this to return `[bundleId, publishedId]` instead, and if you give it published id, it returns `[null, publishedId]`

Reply:
Makes sense, yes I don't think this function is the safest. 
I think this function should eventually take the whole document to check if the `_version` object is present. 

If you have a document with nested ids as you mentioned, it could be understood as a bundle, but it isn't.